### ISSUE TITLE
info : ESET detects meshcentral.exe as coin miner :-(

### ISSUE THREAD

**Issue by tranbert** [Ylianst/MeshCentral#3864]
Since this morning, ESET flags meshcentral.exe as a coin miner...;

![virustotal](https://i.imgur.com/hDBLJmq.png)

The file seems to be Ok with another AV, so I guess^^^hope it's a false positive...

**Comment by Ylianst** [comment]
This has been discussed on [Reddit here](https://www.reddit.com/r/MeshCentral/comments/tzx7ok/win64coinminer_agend/).

The MeshCentral agent certainly does not have any mining code. [The source code to the agent is here](https://github.com/Ylianst/MeshAgent) and the binary is directly compiled from the source on GitHub. I will also add that Bryan and I work for a large company and would get in huge trouble if we pulled anything like that.

Looking at your screen shot, ESET-NOD32 is reporting it wrong, Karpersky, ZoneAlarm and Jiangmin are reporting is correctly as a remote administration tool.

It's not surprising that an AV vendor would get it wrong, just mail then with a false-positive report.

**Comment by tranbert** [comment]
> his has been discussed on [Reddit here](https://www.reddit.com/r/MeshCentral/comments/tzx7ok/win64coinminer_agend/).

Ho, sorry, I missed this one.

> The MeshCentral agent certainly does not have any mining code. [The source code to the agent is here](https://github.com/Ylianst/MeshAgent) and the binary is directly compiled from the source on GitHub. I will also add that Bryan and I work for a large company and would get in huge trouble if we pulled anything like that.

I didn't suspect anything wrong with MeshCentral ! I was afraid of any breach on my MeshCentral  server... I'll send a false positive report to ESET asap...

Thank's

**Comment by Ylianst** [comment]
If multiple AV vendors did report the same thing, I would also be super concerned. However, AV vendors use heuristics and they are a bit all over the map as to what they report. It is true that the MeshAgent is used to do bad things, this is why I do not code sign it. I do code-sign MeshCentral Assistant, but that tool required user opt-in.

**Comment by KommunistGangsta** [comment]
Nice job ESET... Nothing like coming back from the weekend, to find out all of my agents are being deleted....
I'm hoping the ESET server task I made can restore the files... I don't want to reinstall the 80+ agents all over again :(

**Comment by tranbert** [comment]
It Can... I was hopefully using Tactical RMM, so I could reinstall the agent this way. And Tactical uses meshcentral also, and has a built-in agent which may reinstall meshagent also...

But Eset task is another way to do it.

**Comment by KommunistGangsta** [comment]
Well, "lucky" me that I had the ESET Agents installed on all PCs, so I ran a Task on ESET Commander, with the command ("\\srvdeploy\MeshCentral\meshagent64.exe" -fullinstall >> "\\srvdeploy\MeshCentral\logs\%COMPUTERNAME%-Mesh.log")
Even got a small log, so I could verify any errors, or small exceptions where the install failed.

With this I was able to reinstall all the MeshCentral agents.

So those with ESET agents, can quickly solve the problem, by first making an Exclusion for Mesh, and then running the command.

Cheers
